docs/src/miden_library.md

```diff
@@ -0,0 +1,128 @@
+| `execute_foreign_procedure` | Executes the provided procedure against the foreign account.<br><br>Inputs: `[foreign_account_id_prefix, foreign_account_id_suffix, FOREIGN_PROC_ROOT, <inputs>, pad(n)]`<br>Outputs: `[<outputs>]`<br>Context: Any |
+| `update_expiration_block_delta` | Updates the transaction expiration delta.<br><br>Inputs: `[block_height_delta]`<br>Outputs: `[]`<br>Context: Any |
+| `get_expiration_block_delta` | Returns the transaction expiration delta, or 0 if not set.<br><br>Inputs: `[]`<br>Outputs: `[block_height_delta]`<br>Context: Any |
+
+## Faucet Procedures (`miden::faucet`)
+
+Faucet procedures allow reading and writing to faucet accounts to mint and burn assets.
+
+| Procedure | Description |
+| --- | --- |
+| `mint` | Mint an asset from the faucet the transaction is being executed against.<br><br>Inputs: `[ASSET]`<br>Outputs: `[ASSET]`<br>Context: Native & Account & Faucet |
+| `burn` | Burn an asset from the faucet the transaction is being executed against.<br><br>Inputs: `[ASSET]`<br>Outputs: `[ASSET]`<br>Context: Native & Account & Faucet |
+| `get_total_issuance` | Returns the total issuance of the fungible faucet the transaction is being executed against.<br><br>Inputs: `[]`<br>Outputs: `[total_issuance]`<br>Context: Faucet |
+| `is_non_fungible_asset_issued` | Returns a boolean indicating whether the provided non-fungible asset has been already issued by this faucet.<br><br>Inputs: `[ASSET]`<br>Outputs: `[is_issued]`<br>Context: Faucet |
+
+## Asset Procedures (`miden::asset`)
+
+Asset procedures provide utilities for creating fungible and non-fungible assets.
+
+| Procedure | Description |
+| --- | --- |
+| `build_fungible_asset` | Builds a fungible asset for the specified fungible faucet and amount.<br><br>Inputs: `[faucet_id_prefix, faucet_id_suffix, amount]`<br>Outputs: `[ASSET]`<br>Context: Local |
+| `create_fungible_asset` | Creates a fungible asset for the faucet the transaction is being executed against.<br><br>Inputs: `[amount]`<br>Outputs: `[ASSET]`<br>Context: Any |
+| `build_non_fungible_asset` | Builds a non-fungible asset for the specified non-fungible faucet and data hash.<br><br>Inputs: `[faucet_id_prefix, DATA_HASH]`<br>Outputs: `[ASSET]`<br>Context: Local |
+| `create_non_fungible_asset` | Creates a non-fungible asset for the faucet the transaction is being executed against.<br><br>Inputs: `[DATA_HASH]`<br>Outputs: `[ASSET]`<br>Context: Any |
+
+## Implementation Notes
+
+All procedures in the Miden library (except the ones with "Local" context) are implemented as wrappers around underlying kernel procedures. They handle the necessary stack padding and cleanup operations required by the kernel interface, providing a more convenient API for developers.
+
+The procedures maintain the same security and context restrictions as the underlying kernel procedures. When invoking these procedures, ensure that the calling context matches the specified requirements.
```

Comment:
Probably it just me, but I spent a few seconds trying to understand what are the `Implementation Notes`. "Do we have some guide for implementing notes? Or is it refering to the well-known notes? Or there are some notes which implement something?". Well, professional deformation makes itself felt 🫠. I don't think that it will be a problem, but could we maybe somehow rephrase it? But feel free to ignore this.

Also probably it makes sense to move this paragraph up right after the `## Contexts`, so the user will be aware of the usage specific of these procedures before he actually look through them.